src/PlaylistNamespace.js

```diff
@@ -0,0 +1,125 @@
+import assert from 'assert';
+import GMusicNamespace from './GMusicNamespace';
+import Playlist from './Structs/Playlist';
+
+export default class PlaylistNamespace extends GMusicNamespace {
+  constructor(...args) {
+    super(...args);
+    this.path = this._findContextPath();
+    if (this.path) {
+      this._playlists = Object.assign({}, window.APPCONTEXT[this.path[0]][this.path[1]][0][this.path[2]]);
+      this._watchPlaylistObject();
+    } else {
+      this._playlists = {};
+    }
+
+    this.addMethod('getAll', this.getAll.bind(this));
+    this.addMethod('play', this.play.bind(this));
+    this.addMethod('playWithTrack', this.playWithTrack.bind(this));
+  }
+
+  _findContextPath() {
+    let path;
+    Object.keys(window.APPCONTEXT).forEach((key1) => {
+      if (typeof window.APPCONTEXT[key1] === 'object') {
+        const firstLevel = window.APPCONTEXT[key1] || {};
+        Object.keys(firstLevel).forEach((key2) => {
+          if (Array.isArray(firstLevel[key2]) && firstLevel[key2].length > 0) {
+            const secondLevel = firstLevel[key2][0] || {};
+            Object.keys(secondLevel).forEach((key3) => {
```

Comment:
We should use `for` instead of `forEach` as we can't eagerly break out of the loop when our path is found. In the current setup, we continue to iterate in 3 nested for loops which is really bad =X